clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/agent/AwsCertificateManagerCachingAgent.groovy

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2021 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License
+ * for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.aws.provider.agent
+
+import com.amazonaws.services.certificatemanager.AWSCertificateManager
+import com.amazonaws.services.certificatemanager.model.AWSCertificateManagerException
+import com.amazonaws.services.certificatemanager.model.CertificateDetail
+import com.amazonaws.services.certificatemanager.model.CertificateSummary
+import com.amazonaws.services.certificatemanager.model.DescribeCertificateRequest
+import com.amazonaws.services.certificatemanager.model.DescribeCertificateResult
+import com.amazonaws.services.certificatemanager.model.ListCertificatesRequest
+import com.amazonaws.services.certificatemanager.model.ListCertificatesResult
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spectator.api.Id
+import com.netflix.spectator.api.Registry
+import com.netflix.spinnaker.cats.agent.AccountAware
+import com.netflix.spinnaker.cats.agent.AgentDataType
+import com.netflix.spinnaker.cats.agent.CacheResult
+import com.netflix.spinnaker.cats.agent.CachingAgent
+import com.netflix.spinnaker.cats.agent.DefaultCacheResult
+import com.netflix.spinnaker.cats.cache.CacheData
+import com.netflix.spinnaker.cats.cache.DefaultCacheData
+import com.netflix.spinnaker.cats.provider.ProviderCache
+import com.netflix.spinnaker.clouddriver.aws.cache.Keys
+import com.netflix.spinnaker.clouddriver.aws.model.AmazonCertificate
+import com.netflix.spinnaker.clouddriver.aws.provider.AwsInfrastructureProvider
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider
+import com.netflix.spinnaker.clouddriver.aws.security.NetflixAmazonCredentials
+import groovy.util.logging.Slf4j
+import org.apache.commons.lang3.StringUtils
+
+import java.time.Duration
+import java.time.Instant
+
+import static com.netflix.spinnaker.cats.agent.AgentDataType.Authority.AUTHORITATIVE
+import static com.netflix.spinnaker.clouddriver.aws.cache.Keys.Namespace.CERTIFICATES
+
+@Slf4j
+class AwsCertificateManagerCachingAgent implements CachingAgent, AccountAware {
+  final AmazonClientProvider amazonClientProvider
+  final NetflixAmazonCredentials account
+  final String region
+  final ObjectMapper objectMapper
+  final Registry registry
+  final Id securityTokenExceptionGauge
+
+  protected Instant lastFailure
+
+  static final Set<AgentDataType> types = Collections.unmodifiableSet([
+    AUTHORITATIVE.forType(CERTIFICATES.ns)
```

Comment:
So looking at https://github.com/spinnaker/clouddriver/blob/f3b393402c603f5b19f7cc10ac88d137269b702c/clouddriver-api/src/main/java/com/netflix/spinnaker/cats/agent/AgentDataType.java#L24 it seems like if this is listing itself as `AUTHORITATIVE` then it's going to fight with https://github.com/spinnaker/clouddriver/blob/master/clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/agent/AmazonCertificateCachingAgent.groovy over which agent has the complete list.

I think it might be better to add the reading from Certificate Manager to the `AmazonCertificateCachingAgent` so that there's only one agent.